lib/java_buildpack/framework/snyk.rb

```diff
@@ -0,0 +1,293 @@
+        request['Authorization'] = "token #{api_token}"
+
+        return uri, request
+      end
+
+      def do_request(uri, request)
+        https = Net::HTTP.new(uri.host, uri.port)
+        https.use_ssl = true
+        response = https.request(request)
+        if !response.is_a? Net::HTTPSuccess
+          raise "HTTP error #{ response.code } #{ response.message }"
+        end
+        return response
+      end
+
+      def gather_pom_files
+        target_pom_path = find_main_pom
+        target_pom = target_pom_path ? File.read(target_pom_path) : ""
+        additional_poms = all_jars.map{ |jar| poms_from_jar(jar) }.flatten
+        
+        if not target_pom.empty?
+          return target_pom, additional_poms
+        elsif not additional_poms.empty?
+          return additional_poms[0], additional_poms[1..-1]
+        else
+          @logger.warn "no .pom files found".yellow
+          return "", []
+        end
+      end
+
+      def find_main_pom
+        Dir.glob("#{@droplet.root}/**/pom.xml")[0]
+      end
+
+      def all_jars
+        Dir.glob("#{@droplet.root}/WEB-INF/**/*.jar")
+      end
+
+      def poms_from_jar(jar)
+        ret = []
+        poms = `unzip -Z1 #{jar} | grep "pom.xml"`.split('\n')
+        poms.map{ |pom| `unzip -p #{jar} #{pom}` }
+      end
+
+      def print_results(test)
+        @logger.info " "
+        @logger.info "Testing #{@application.details['application_name']}...\n".white.bold
+        @logger.info " "
+
+        issues = []
+        issues.concat(test['vulnerabilities']) if test.key?('vulnerabilities')
+        issues.concat(test['issues']['vulnerabilities']) if (test.key?('issues') and test['issues'].key?('vulnerabilities'))
+        issues.concat(test['issues']['licenses']) if (test.key?('issues') and test['issues'].key?('licenses'))
+        issues = sort_issues(issues)
+        if severity_threshold
+          issues = filter_issues(issues, severity_threshold)
+        end
+        
+        issues.each do |issue|
+          print_issue(issue)
+        end
+
+        # puts "#{'Organisation:'.bold}      #{test['org']['name']}"
+        # puts "#{'Package manager:'.bold}   #{test['packageManager']}"
+        # puts "#{'Target file:'.bold}       -"
+        # puts "#{'Open source:'.bold}       ?"
+        # puts "#{'Project path:'.bold}      ?"
+
+        dependencyCount = test['dependencyCount'] || 0
+        if issues.length == 0
+          @logger.info "✓ Tested #{dependencyCount} dependencies for known vulnerabilities, no vulnerable paths found\n".green
+        else
+          unique_count = issues.map{ |issue| issue['id']}.uniq.length
+          vulnerable_paths = issues.map{ |issue| issue['from']}.flatten.uniq.length
+          @logger.info "\e[0mTested #{dependencyCount} dependencies for known vulnerabilities, #{"found #{unique_count} vulnerabilities, #{vulnerable_paths} vulnerable paths.".red}\n"
+        end
+        @logger.info " "
```

Comment:
Why do we need this? \n is not enough?

Reply:
I changed everything to " ". Using "\n" scrambled the logger output